app/components/blacklight/hierarchy/qfacet_value_component.rb

```diff
@@ -14,7 +14,14 @@ def initialize(field_name:, item:, id: nil, suppress_link: false)
 
       def path_for_facet
         facet_config = helpers.facet_configuration_for_field(field_name)
-        Blacklight::FacetItemPresenter.new(item.qvalue, facet_config, helpers, field_name).href
+        presenter_class = if helpers.search_state.params.except(:controller, :action).present?
+           Blacklight::FacetItemPresenter
+        else
+          # Bypasses slow Blacklight machinery for empty search state.
```

Comment:
Which code in particular is slow? Can you point at it?

Reply:
I wish I could. I benchmarked all the way to the bottom of Blacklight and it is the cumulative impact of the state / parameter munging.